File: ReadyTasks/FSharp/ProjectEuler/Task_47/Task_47/Program.fs
open System

let getPrimeDivsCount numb =
    let rec getDiv numb div =
        if (numb % div > 0) then
            numb
        else
            getDiv (numb / div) div

    let rec helpFunc numb ind =
        if (numb = 1) then
            0
        else if (ind * ind > numb) then
            1
        else if (numb % ind = 0) then
            let newNumb = getDiv numb ind
            1 + helpFunc newNumb (ind + 1)
        else
            helpFunc numb (ind + 1)
    helpFunc numb 2
        
let getResult numb =
    let filterFunc n =
        [n..(n + numb - 1)] |> List.forall (fun x -> getPrimeDivsCount x = numb)
    Seq.initInfinite (fun x -> x + 1) |> Seq.filter filterFunc

[<EntryPoint>]
let main argv =
    printfn "%A" (getResult 4)
    0 // return an integer exit code

File: ReadyTasks/CSharp/ProjectEuler/Task_47/Task_47/Program.cs
using System;

namespace Task_47
{
    class Program
    {
        static int CountDivs(int numb)
        {
            int ind = 2;
            int count = 0;
            while (numb > 1)
            {
                if (numb % ind == 0)
                {
                    count++;
                    while (numb % ind == 0)
                    {
                        numb /= ind;
                    }
                }
                ind++;
            }
            return count;
        }

        static bool CheckProperty(int numb, int k)
        {
            for (int i = 0; i < k; i++)
            {
                if (CountDivs(numb + i) != k)
                {
                    return false;
                }
            }
            return true;
        }

        static int GetResult(int numb)
        {
            for (int i = 1; ; i++)
            {
                if (CheckProperty(i, numb))
                {
                    return i;
                }
            }
            return -1;
        }

        static void Main(string[] args)
        {
            Console.WriteLine(GetResult(4));
        }
    }
}
